### PR TITLE
[Codegen][CUDA] Fix: cuda codegen vectorize cast

### PR DESCRIPTION
**Pull request by kongroo**
Data types such as float32x8 and int32x8 are not supported in CUDA, which will result in errors like "Cannot convert type float32x8 to CUDA type!" in code generation. I tried to fix this by storing 2 32-bits values in 1 64-bits value.
Could you please help review this fix? @Laurawly

### CODE REVIEW

File: src/target/source/codegen_cuda.cc [hunk 1]
```diff
@@ -238,12 +253,53 @@ void CodeGenCUDA::PrintType(DataType t, std::ostream& os) {  // NOLINT(*)
...
+          // s4.w is emitted as *(short2*)(&(i2.y)).y
+          //
+          ICHECK_EQ(t.lanes() % 2, 0) << "only support even lane for shorT type with lanes > 4";
+          os << "int" << t.lanes() / 2;
+        } else {
+          fail = true;
+        }
+        if (!fail) {
+          return;
+        }
```
Comment (Laurawly) [inline src/target/source/codegen_cuda.cc:278]: Missing `break` here?

File: tests/python/unittest/test_target_codegen_cuda.py [hunk 2]
```diff
@@ -511,8 +511,8 @@ def check(t0, t1):
 
         # schedule
         s = tvm.te.create_schedule(C.op)
-        ob, ib = s[C].split(s[C].op.axis[0], nparts=32)
-        _, iib = s[C].split(ib, factor=4)
+        ob, ib = s[C].split(s[C].op.axis[0], nparts=n // factor)
```
Comment (Laurawly) [inline tests/python/unittest/test_target_codegen_cuda.py:514]: We can also directly say `factor=factor` here.

**Review by Laurawly — COMMENTED** [review]
Just a few comments. Also cc @wpan11nv @vinx13 .

File: tests/python/unittest/test_target_codegen_cuda.py (hunk 2 above)
Comment (kongroo) [reply tests/python/unittest/test_target_codegen_cuda.py:514]: like this?
```python
ob, ib = s[C].split(s[C].op.axis[0], factor=factor)
# _, iib = s[C].split(ib, factor=factor)
s[C].vectorize(ib)
```

Comment (Laurawly) [reply tests/python/unittest/test_target_codegen_cuda.py:514]: Yeah.

File: src/target/source/codegen_cuda.cc (hunk 1 above)
Comment (kongroo) [reply src/target/source/codegen_cuda.cc:278]: fixed

File: tests/python/unittest/test_target_codegen_cuda.py (hunk 2 above)
Comment (kongroo) [reply tests/python/unittest/test_target_codegen_cuda.py]: changed

**Review by vinx13 — APPROVED** [review]

**Comment by vinx13** [comment]
Thanks @kongroo @Laurawly

**Pull request merged**